data/src/main/java/com/microsoft/azure/kusto/data/auth/endpoints/WellKnownKustoEndpointsData.java

```diff
@@ -0,0 +1,46 @@
+package com.microsoft.azure.kusto.data.auth.endpoints;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class WellKnownKustoEndpointsData {
+    public static class AllowedEndpoints {
+        public ArrayList<String> AllowedKustoSuffixes;
+        public ArrayList<String> AllowedKustoHostnames;
+    }
+
+    @JsonIgnore
+    public ArrayList<String> _Comments;
+    public HashMap<String, AllowedEndpoints> AllowedEndpointsByLogin;
+    private static WellKnownKustoEndpointsData instance = null;
+
+    public static WellKnownKustoEndpointsData getInstance() {
+        if (instance == null)
+            instance = readInstance();
+
+        return instance;
+    }
+
+    // For Deserialization
+    public WellKnownKustoEndpointsData() {
+    }
+
+    private static WellKnownKustoEndpointsData readInstance() {
+        try {
+            // Beautiful !
+            ObjectMapper objectMapper = new ObjectMapper();
+            try (InputStream resourceAsStream = WellKnownKustoEndpointsData.class.getResourceAsStream(
```

Comment:
Make sure it's compiled into the jar and doesn't need to exist on the filesystem

Reply:
its a resource